app/controllers/experiments_controller.rb

```diff
@@ -380,11 +378,35 @@ def create
         replication_level: [:optional, :security_default, :integer, :positive],
         execution_time_constraint: [:optional, :security_default, :integer, :positive],
         parameter_constraints: [:optional, :security_json],
-        type: [:optional, :security_default]
+        type: [:optional, :security_default],
+        workers_scaling: [:optional] # TODO boolean validator
+        #workers_scaling_params: [:optional, :json_or_hash] TODO uncomment? SCAL-757: (..) json_or_hash validator bug
     )
     type = params[:type] || 'experiment'
-    raise ValidationError.new('type', type, 'Not a correct experiment type') unless CONSTRUCTORS.has_key? type
-    send(CONSTRUCTORS[type])
+    Utils::raise_error_unless_has_key(CONSTRUCTORS, type, "Not a correct experiment type: #{type}")
+    workers_scaling_params = nil
+    if params.has_key? :workers_scaling and params[:workers_scaling]
+      Utils::raise_error_unless_has_key(params, :workers_scaling_params, 'Missing workers scaling parameters')
+      workers_scaling_params = Utils::parse_json_if_string(params[:workers_scaling_params]).symbolize_keys
+      message_prefix = 'Missing workers scaling parameter'
+      Utils::raise_error_unless_has_key(workers_scaling_params, :name, "#{message_prefix} name",
+                                        'workers_scaling_params')
+      Utils::raise_error_unless_has_key(workers_scaling_params, :allowed_infrastructures,
+                                        "#{message_prefix} allowed_infrastructures", 'workers_scaling_params')
+      Utils::raise_error_unless_has_key(workers_scaling_params, :time_limit, "#{message_prefix} time_limit",
+                                        'workers_scaling_params')
+      # TODO more precise validation
+    end
+    experiment = send(CONSTRUCTORS[type])
+    if params.has_key? :workers_scaling and params[:workers_scaling]
```

Comment:
zostawienie samego
```
if params[:workers_scaling]
```
zadziała tak samo, jeżeli nie ma takiego klucza to wróci nil, który jest falsy. Z drugiej strony nie wiem czy to będzie wystarczająco czytelne.